websocket/src/websocket.cpp

```diff
@@ -771,40 +816,33 @@ static dmExtension::Result OnUpdate(dmExtension::Params* params)
             }
 
 #if defined(__EMSCRIPTEN__)
-            conn->m_SSLSocket = dmSSLSocket::INVALID_SOCKET_HANDLE;
-
-            EM_ASM({
-                // https://emscripten.org/docs/porting/networking.html#emulated-posix-tcp-sockets-over-websockets
-                Module["websocket"]["subprotocol"] = $0 ? UTF8ToString($0) : null;
-            }, conn->m_Protocol);
-
             char uri_buffer[dmURI::MAX_URI_LEN];
             const char* uri;
             if (conn->m_Url.m_Path[0] != '\0') {
-                dmSnPrintf(uri_buffer, sizeof(uri_buffer), "%s%s", conn->m_Url.m_Hostname, conn->m_Url.m_Path);
-                uri = uri_buffer;
+                dmSnPrintf(uri_buffer, sizeof(uri_buffer), "%s://%s%s", conn->m_Url.m_Scheme, conn->m_Url.m_Hostname, conn->m_Url.m_Path);
             } else {
-                uri = conn->m_Url.m_Hostname;
-            }
-
-            dmSocket::Address address;
-            dmSocket::Result sr = dmSocket::GetHostByName(uri, &address, true, false);
-            if (dmSocket::RESULT_OK != sr) {
-                CLOSE_CONN("Failed to get address from host name '%s': %s", uri, dmSocket::ResultToString(sr));
-                continue;
+                dmSnPrintf(uri_buffer, sizeof(uri_buffer), "%s://%s", conn->m_Url.m_Scheme, conn->m_Url.m_Hostname);
             }
-
-            sr = dmSocket::New(address.m_family, dmSocket::TYPE_STREAM, dmSocket::PROTOCOL_TCP, &conn->m_Socket);
-            if (dmSocket::RESULT_OK != sr) {
-                CLOSE_CONN("Failed to create socket for '%s': %s", conn->m_Url.m_Hostname, dmSocket::ResultToString(sr));
+            uri = uri_buffer;
+
+            EmscriptenWebSocketCreateAttributes ws_attrs = {
+                uri,
+                conn->m_Protocol,
+                EM_TRUE
```

Comment:
Not entirely sure here. Docs say:

If true, the created socket will reside on the main browser thread. If false, the created socket is bound to the calling thread. If you want to share the created EMSCRIPTEN_WEBSOCKET_T structure across multiple threads, or are running your own main loop in the pthread that you create the socket, set createOnMainThread to true. If the created WebSocket only needs to be accessible on the thread that created it, and the creating thread is an event based thread (meaning it regularly yields back to the browser event loop), then it is more efficient to set this to false.